workflow_tests/README.md

```diff
@@ -0,0 +1,422 @@
+```javascript
+let act = new kieActJs.Act(repoPath, workflowPath);
+```
+Set up initial parameters for `Act`. This is where we can set secrets, GitHub token and options for the events (like the name of the branch to which the push has been made, etc.)
+```javascript
+act = utils.setUpActParams(
+    act,
+    '<event>',
+    {head: {ref: '<branch_name>'}},
+    {'<SECRET_NAME>': '<secret_value'},
+    '<github_token>',
+);
+```
+Set up step mocks. Here we configure which steps in the workflow should be mocked, and with what behaviour. This takes form of an object with keys corresponding to the names of the jobs in the workflow, and values being mock definitions for specific steps. The steps can be identified either by `id`, `name`, `uses` or `run`. Step mock can be defined either by hand (`<job_1_name>`) or with the helper method `utils.getMockStep()` (`<job_2_name>`). Not mocked steps will be executed normally - **make sure this will not have unexpected consequences**
+```javascript
+const testMockSteps = {
+    '<job_1_name>': [
+        {
+            name: '<step_1_1_name>',
+            mockWith: '<mock_command>',
+        },
+        {
+            name: '<step_1_2_name>',
+            mockWith: '<mock_command>',
+        },
+    ],
+    '<job_2_name>': [
+        utils.getMockStep('<step_2_1_name>', '<message>'),
+        utils.getMockStep('<step_2_2_name>', '<message>'),
+    ],
+};
+```
+Most important part - actually running the event with `Act`. This executes the specified `<event>` in the context of the local test repo created before and with the workflow under test set up. `result` stores the output of `Act` execution, which can then be compared to what was expected. Note that the `workflowFile` is actually path to _workflow folder_ and not the file itself - `Act-js` determines the name of the workflow by itself, and tries to find it in the specified `workflowFile` path, so _providing the full path to the file will fail_
+```javascript
+const result = await act
+    .runEvent('<event>', {
+        workflowFile: path.join(repoPath, '.github', 'workflows'),
+        mockSteps: testMockSteps,
+    });
+```
+Assert results are as expected. This can, for example, include using `expect()` to check if the steps that should be executed have indeed been executed, steps that shouldn't run have not been executed, compare statuses (which steps succeeded, which failed) and step outputs. Outputs can include additional information, like input values, environmental variables, secrets (although these are usually not accessible and represented by `***`, this can still be useful to check if the value exists or not). Here it's usually done with the helper assertion methods defined in the assertions file. Step assertions can be created manually or with `getStepAssertion()` helper method
+```javascript
+assertions.assertSomethingHappend(result);
```

Comment:
```suggestion
assertions.assertSomethingHappened(result);
```

Reply:
Updated in https://github.com/Expensify/App/pull/21937